components/session-based-conditional-auth-functions/src/main/java/org/wso2/carbon/identity/extensions/authenticator/conditional/auth/functions/function/KillSessionFunction.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.wso2.carbon.identity.extensions.authenticator.conditional.auth.functions.function;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.authentication.framework.config.model.graph.js.JsAuthenticationContext;
+import org.wso2.carbon.identity.application.authentication.framework.services.SessionManagementService;
+import org.wso2.carbon.identity.extensions.authenticator.conditional.auth.functions.util.SessionValidationConstants;
+
+import java.util.Map;
+
+/**
+ * Represents javascript function provided in conditional authentication to terminate a session with given sessionID
+ */
+public class KillSessionFunction implements ExecuteActionFunction {
+
+    private static final Log log = LogFactory.getLog(KillSessionFunction.class);
+
+    @Override
+    public Boolean execute(JsAuthenticationContext context, Map<String, String> map) {
+
+        String sessionId = map.get(SessionValidationConstants.TERMINATION_SESSION_ID);
+        try {
+            SessionManagementService sessionManagementService = new SessionManagementService();
+            if (log.isDebugEnabled()) {
+                log.debug("Session with session id :" + sessionId + " is requested to kill");
+            }
+            sessionManagementService.removeSession(sessionId);
+        } catch (NullPointerException e) {
```

Comment:
NullPointerException is a run-time exception which is not recommended to catch it. Is there any reason to catch the  runtime exception in here.

Reply:
In here session ID is taken from the conditional authentication script. So it possible that it is an invalid session ID. That is why I thought it is better to catch it and return false instead of throwing it

Reply:
If there is a possible chance, of NPE, then it should be fixed by checking whether particular sessionId is null. Catching NullPointer is not correct way to handle it.

Reply:
problem is not the sessionID being null. A user may be asking to kill a session with a given sessionID. But there may not be a session with that session ID or that session may have expired already. In that case sessionManagementService.removeSession(sessionId) throws a NPE(). IMO that should not be thrown as an exception. Instead we should indicate that we failed to remove that session
Refer this for SessionManagementService implementation
https://github.com/wso2/carbon-identity-framework/blob/master/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/services/SessionManagementService.java